### PR TITLE
samples: add iio/proximity-sensor sample

### PR DESCRIPTION
**Pull request by yongli3**
Signed-off-by: Yong Li <yong.b.li@intel.com>

### CODE REVIEW

**Comment by yongli3** [comment]
@bdilly @neofang @peijiajames this PR is used for issue #2207, please help to review

**Comment by bdilly** [comment]
Hi @yongli3 , it seems good.

**Comment by edersondisouza** [comment]
:+1:

File: src/samples/flow/iio/sol-flow-intel-minnow-max-linux_gt_3_17.json [hunk 1]
```diff
@@ -64,6 +64,14 @@
            "buffer_size": "-1"
      },
      "type": "iio/thermometer"
+  },
+  {
+     "name": "MyProximity",
+     "options": {
+           "iio_device": "apds9960 create,i2c,platform/80860F41:05,0x39,apds9960 0",
+           "buffer_size": "-1"
```
Comment (barbieri) [inline src/samples/flow/iio/sol-flow-intel-minnow-max-linux_gt_3_17.json]: any reason why disable buffering?

Comment (yongli3) [inline src/samples/flow/iio/sol-flow-intel-minnow-max-linux_gt_3_17.json]: @barbieri I found not all sensors support the buffering/trigger reading. In this case, the apds9960 does not support it

Comment (barbieri) [inline src/samples/flow/iio/sol-flow-intel-minnow-max-linux_gt_3_17.json]: @edersondisouza explained to me at IRC, but I think the "0" (auto) logic is missing some pieces and should understand devices that do not support buffering. We'll fix that later, after that we could always go with the default value (0) and it should work, the buffer must be used only explicitly.

Comment (yongli3) [inline src/samples/flow/iio/sol-flow-intel-minnow-max-linux_gt_3_17.json]: I got your point. Curently the “buffer_size=0” will use the default buffer size, it does not fallback to “normal reading” if the driver does not support buffering. We need to enhance the logic, to check if it supports buffering when “buffer_size > 0”

**Pull request merged**